fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -885,6 +891,18 @@ public void analyze(TQueryOptions tQueryOptions) throws UserException {
             if (parsedStmt instanceof QueryStmt) {
                 queryStmt = (QueryStmt) parsedStmt;
                 queryStmt.getTables(analyzer, false, tableMap, parentViewNameSet);
+
+                // queue query here
+                if (Config.enable_resource_group && Config.enable_query_queue) {
+                    this.queryQueue = analyzer.getEnv().getResourceGroupMgr()
+                            .getResourceGroupQueryQueue(context.sessionVariable.resourceGroup);
+                    try {
+                        queryQueue.offer();
```

Comment:
Stuck here, will the user be reminded, such as "query is in queue"

Reply:
Sure, it's necessary; We can using ```Show query stmt``` to achive this.
But it's a todo.